nautobot/dcim/forms.py

```diff
@@ -3803,6 +3803,14 @@ class ConnectCableToDeviceForm(ConnectCableExcludeIDMixin, NautobotModelForm):
             "rack": "$termination_b_rack",
         },
     )
+    termination_b_module = DynamicModelChoiceField(
+        queryset=Module.objects.all(),
+        label="Module",
+        required=False,
+        query_params={
+            "device": "$termination_b_device",
```

Comment:
Can/should we filter by `$termination_b_location` as well? For non-deviced modules?

Reply:
I tried that but it didn't work since the the `device` and `location` filters are ANDed together and there's never a case where both of those fields will be populated. I could do a custom filter for this purpose to OR those two fields together but I didn't think there was a use case for cabling up a module in storage (not installed in a device)

Reply:
Would it make sense to enhance the module `location` filter to include `device__location` as part of its capabilities? Fine if the answer is no.

Reply:
Another option is to use a custom method filter for `location` and make it filter on either the parent device's  location or the module's `location` field but it might be difficult to write an efficient query for that.

Reply:
The `location` filter would need to be a method filter so we could look up nested modules' locations (`parent_module_bay__device | parent_module_bay__parent_module__parent_module_bay__device` etc). It's not a problem to create that filter if we need it.